### PR TITLE
Translation of player and game classes

### PR DESCRIPTION
**Pull request by nd1az**
I did .ts translation on player component and game component.
I also generated an angular component and module for the chat component so that translation could be push there
Work on updating the map class to suit interacting with the game class will be done later tonight.

### CODE REVIEW

File: appGeo/src/app/game/game.component.ts [hunk 1]
```diff
@@ -1,15 +1,142 @@
...
+              count += 1
+          }
+      }
+
+      return count
+
+  }
+
+  getFractionRole(countKiller) {
+      return (this.countRole(countKiller) / this.PlayerCount)  //does this use RoleCount? the names are different
```
Comment (noahklowden) [inline appGeo/src/app/game/game.component.ts:113]: this should be getRoleCount()

File: appGeo/src/app/game/game.component.ts [hunk 2]
```diff
@@ -1,15 +1,142 @@
...
+  }
+
+  getSnapshot(snapshotID) {
+      return this.snapshots.get(snapshotID)
+  }
+
+  addSnapshot(snapshot) {
+      this.snapshots.add(snapshot.getUserID(), snapshot)
```
Comment (noahklowden) [inline appGeo/src/app/game/game.component.ts:121]: Map uses set instead of add

File: appGeo/src/app/game/game.component.ts (hunk 1 above)
Comment (noahklowden) [inline appGeo/src/app/game/game.component.ts:113]: this should be getRoleCount()

File: appGeo/src/app/game/game.component.ts [hunk 3]
```diff
@@ -1,15 +1,142 @@
...
+  removeSnapshot(snapshotID) {
+      return this.snapshots.delete(snapshotID)
+  }
+
+  getChat(chatID) {
+    return this.chats.get(chatID)
+  }
+
+  addChat(chat) {
+    this.chats.add(chat.getUserID(), chat)
```
Comment (noahklowden) [inline appGeo/src/app/game/game.component.ts:133]: Map uses set instead of add

File: appGeo/src/app/game/game.component.ts [hunk 4]
```diff
@@ -1,15 +1,142 @@
...
+
+  getRoleCount(countKiller) {
+
+      const playersList = this.map[Symbol.iterator]();//need to adjust the code for Campus Map for this to work
+
+
+      var count = 0
+
+      for(var p in playersList) {
+          if(!(typeof p == Killer) ^ (countKiller)) {
```
Comment (noahklowden) [inline appGeo/src/app/game/game.component.ts:103]: This was poorly written by me originally. But should be swapped to
`if((playersList[i] instanceof Killer) && (countKiller) || 
               !(playersList[i] instanceof Killer) && !(countKiller))`
to fit in typescript

File: appGeo/src/app/game/game.component.js [hunk 5]
```diff
@@ -3,7 +3,10 @@ const ACTIVE = 1
 const CIVILIAN = 0
 const KILLER = 1
 
-import {Killer} from 'src/app/map/player.component.js';
+import {Player, Civillian, Killer} from '../player/player.component.js';
+import {CampusMap} from '../map/campus-map.component.js'
+//A CampusMap is a Map of the Bubbles that exist on campus
+
```
Comment (noahklowden) [inline appGeo/src/app/game/game.component.js:9]: Chat should be imported somewhere here

File: appGeo/src/app/player/player.component.js [hunk 6]
```diff
@@ -1,3 +1,7 @@
+import{Message} from '../chat/chat_class_declaration.js';
+import{Chat} from '../chat/chat_class_declaration.js';
+import{Location} from './location_class_declaration.js';
+
```
Comment (noahklowden) [inline appGeo/src/app/player/player.component.js:4]: Chat and Message should be imported here

File: appGeo/src/app/player/player.component.ts [hunk 7]
```diff
@@ -1,111 +1,263 @@
...
-
-
-  ngOnInit(): void {
-  }
-
-}
+import { Component, OnInit } from '@angular/core';
+import { InjectableAnimationEngine } from '@nativescript/angular';
+import { ChangeType } from '@nativescript/core';
+import { Bubble } from '../map/map.component';
```
Comment (noahklowden) [inline appGeo/src/app/player/player.component.ts:4]: Chat and Message should be imported here

**Review by noahklowden — CHANGES_REQUESTED** [review]
Mostly this looks good. However, there are some compiler issues (especially a few issues in how I wrote the original code for game that I hadn't merged the fixes for yet) plus a few imports missing.

File: appGeo/src/app/game/game.component.ts (hunk 4 above)
Comment (nd1az) [reply appGeo/src/app/game/game.component.ts]: I adjusted it according to your suggestion. Hope it looks as intended now

**Comment by nd1az** [comment]
I notice some errors come from chat not being translated into .ts yet as well since playercomponent can't compile properly without the message class being exported with the right constructor

**Comment by noahklowden** [comment]
> I notice some errors come from chat not being translated into .ts yet as well since playercomponent can't compile properly without the message class being exported with the right constructor

This does seem like it can be resolved with @jchee1's updates. The question is just which PR should be done first

**Review by noahklowden — APPROVED** [review]
Looks good!

**Comment by nd1az** [comment]
Saw that the chat.component.spec that generated with the component files caused the test run to fail. I commented it out so that the pr can go through

**Pull request merged**